### PR TITLE
feat(table): added sticky-header variation

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly/issues/2838

### CODE REVIEW

**Comment by redallen** [comment]
I'll try to rebase this on `master` momentarily.

**Comment by redallen** [comment]
Rebased, let me know if anything is amiss.

**Comment by mcoker** [comment]
Thanks @redallen looks perfect!

**Comment by KendraMar** [comment]
@evwilkin This works great for the Insights apps on cloud.redhat.com. Is it possible to get this incorporated into the next PatternFly milestone?

@katierik 
@mcarrano

**Comment by evwilkin** [comment]
@KendraMar it looks straightforward but I'll raise it at our next PF meeting and make sure there's nothing we're missing to get it slotted in 👍

File: src/patternfly/components/Table/table.scss [hunk 1]
```diff
@@ -200,6 +200,31 @@
...
+    position: relative;
+
+    > thead > tr {
+      border-bottom: 0;
+
+      > * {
+        position: sticky;
+        top: 0;
+        z-index: var(--pf-global--ZIndex--xs);
+        background: var(--pf-c-table--BackgroundColor);
```
Comment (christiemolloy) [inline src/patternfly/components/Table/table.scss:213]: why is this necessary?

Comment (mcoker) [reply src/patternfly/components/Table/table.scss:213]: it gives the header cells a background so that the contents of the table don't show through as you scroll
Jun-17-2020 15-13-48

Comment (christiemolloy) [reply src/patternfly/components/Table/table.scss:213]: oh awesome!

**Comment by evwilkin** [comment]
@KendraMar FYI we've got this slotted to wrap up for the next milestone: https://github.com/patternfly/patternfly-react/issues/4406
